docs/specification/draft/basic/transports.mdx

```diff
@@ -287,6 +288,129 @@ protocol version 2024-11-05) as follows:
+- Server maintains session state and processes requests normally
+
+#### Unintended Disconnection Handling
+
+WebSocket connection interruption might occur at any time (e.g., due to network conditions). Therefore:
+- WebSocket connection interruption **MUST NOT** be interpreted as the client cancelling an ongoing request
+- To cancel, the client **SHOULD** explicitly send an MCP `CancelledNotification`
+- Similarly, disconnection **MUST NOT** be interpreted as terminating the MCP session
+- Server **MUST** preserve the state of any sessions associated with this connection for a reasonable timeout period to allow reconnection
+
+#### Reconnection and Session Recovery
+
+To reconnect:
+- Client **MAY** include one `mcp.session-id.${mcpSessionId}` value in the WebSocket subprotocols list via the `Sec-WebSocket-Protocol` header in the WebSocket upgrade request to help with intermediary affinity
+- Client **MUST** include `mcp.auth.${token}` in the WebSocket subprotocols list
+- Server **MUST** validate auth before proceeding. Auth is associated with the connection, not a particular session
+- Client **MUST** include the `mcpSessionId` from the previous session in any message after reconnection
+- Client **MAY** send an `initialize` message including a `params.mcpSessionId` after reconnecting over a different WebSocket connection to an existing session
+  - Server **MUST** associate this new connection with that session and update any renegotiated version or capabilities
+- Regardless, upon receipt of the first message for an existing session over a different connection, the server **MUST** associate this new connection with previously established session state making this the only active connection for the session
+- Server **MUST** disassociate any previous connection associated with the same `mcpSessionId` and ignore any messages that arrive for the MCP session on any connection but the active one
+- Server **SHOULD** keep track of recently disassociated connection-session pairs to prevent thrashing of a session between two connections
+- If no other sessions are active on the recently disassociated connection, the server **SHOULD** close that connection with a `1000` "Normal Closure" status code
+- Clients and servers **SHOULD** keep track of unacknowledged requests and resend them if necessary after reconnection
+- Normal operation can resume immediately after MCP session recognition
+
+#### Intentional Connection Close Handling
+
+- The server or client **MAY** close the WebSocket connection at any time by sending a close frame with an appropriate close code and reason.
+- Session termination is independent of connection. So, the server **SHOULD NOT** automatically terminate sessions associated with the connection being closed.
+
+### Error Handling
+
+#### HTTP to WS Upgrade Errors
+
+Servers **MUST** return appropriate HTTP status codes during WebSocket upgrade failures:
+
+- `404 Not Found`: WebSocket endpoint or message handler not found
```

Comment:
Message handler not found?

Reply:
Good catch. That's an uncleaned up artifact of an earlier version where I had multiple things that could be done on the endpoint. I'll remove it but maybe not before 11am.